modules/core/src/main/java/org/apache/ignite/internal/util/IgniteUtils.java

```diff
@@ -8720,6 +8720,31 @@ public static String fl(String msg, Object... args) {
         return sb.toString();
     }
 
+    /**
+     * Round up the argument to the next highest power of 2;
+     *
+     * @param v Value to round up.
+     * @return Next closest power of 2.
+     */
+    public static int nextPowerOf2(int v) {
```

Comment:
`Integer.highestOneBit(value) << 1` ?

Reply:
Not quite because it does not check if the integer is already is a power of 2.
Replaced with 1 << (32 - Integer.numberOfLeadingZeros(v - 1))